### PR TITLE
Migrate to 2018 Edition

### PR DESCRIPTION
**Pull request by ghost**
This is a first attempt to migrate to the 2018 Edition. There didn't appear to be any complications.

This is intended to address #67.

### CODE REVIEW

**Comment by ghost** [comment]
I'm not encountering the nightly build errors on my machine. I'll investigate later.

**Comment by porglezomp** [comment]
Ah, it looks like I've got CI set up to build on stable, not nightly, so I'll need to fix that before this can be merged. Thank you for the PR!

**Comment by porglezomp** [comment]
If you merge or rebase onto develop, that should get this passing CI.

File: Cargo.toml [hunk 1]
```diff
@@ -8,6 +8,8 @@ readme = "README.md"
 keywords = ["comic", "rss", "atom", "syndication"]
 categories = ["command-line-utilities"]
 publish = true
+cargo-features = ["edition"]
```
Comment (porglezomp) [inline Cargo.toml]: This isn't needed on the latest nightly.

**Review by porglezomp — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
Interesting that it's erroring on Linux and Windows, but not macOS.

**Comment by porglezomp** [comment]
Ah, that's because `cargo fix` only runs on the enabled modules, you'll need to manually do the fixes in the other platform modules.

**Comment by ghost** [comment]
> Ah, that's because `cargo fix` only runs on the enabled modules, you'll need to manually do the fixes in the other platform modules.

That is what I am attempting to do.

**Comment by porglezomp** [comment]
Yeah, right after I said that I noticed you had done that.

**Comment by ghost** [comment]
Appveyor appears to be trying to build on stable, despite your changes.

**Comment by porglezomp** [comment]
You're right, can you change in the `install` section to in `appveyor.yml`:
```diff
-  - rustup-init.exe -y --default-host %TARGET%
+  - rustup-init.exe -y --default-host %TARGET% --default-toolchain %CHANNEL%
```

**Comment by porglezomp** [comment]
I *think* the last missing module is the `platform::linux`.

File: src/main.rs [hunk 2]
```diff
@@ -1,12 +1,8 @@
...
 #[macro_use]
 extern crate log;
-extern crate pretty_env_logger;
-extern crate regex;
-extern crate reqwest;
-extern crate syndication;
-extern crate xdg;
+use pretty_env_logger;
+
+use reqwest;
```
Comment (pickfire) [inline src/main.rs:5]: Is `pretty_env_logger` and `reqwest` still needed?

**Comment by porglezomp** [comment]
@LEARAX I believe the only thing that remains to be done on this is the fix the `platform::linux` module. Are you able to finish this up?

**Comment by ghost** [comment]
@porglezomp CI passed.

**Review by porglezomp — APPROVED** [review]

**Comment by porglezomp** [comment]
Thanks for your contribution! 🎉 ✨ 🎉 🎆

**Pull request merged**